### PR TITLE
Multiband

### PR DESCRIPTION
**Pull request by b-biswas**
Adding multi-band processing 
- Shears are measured individually in bands

### CODE REVIEW

File: deep_field_metadetect/jaxify/jax_detection.py [hunk 1]
```diff
@@ -637,3 +637,45 @@ def scan_fn(carry, xy):
...
+    coadd_im = jnp.zeros_like(first_obs.image)
+    coadd_noise = jnp.zeros_like(first_obs.image)
+    mask = jnp.zeros(coadd_im.shape, dtype=jnp.int32)
+    total_weight = jnp.zeros_like(first_obs.image)
+
+    for obs in obs_list:
+        coadd_im = coadd_im + obs.image * obs.weight
+        coadd_noise = coadd_noise + obs.noise * obs.weight
+        total_weight = total_weight + obs.weight
+        mask = mask | obs.bmask
```
Comment (beckermr) [inline deep_field_metadetect/jaxify/jax_detection.py:674]: We should be sure the weight is the same for every pixel. So we'll want a median of the weight map.

File: deep_field_metadetect/jaxify/jax_detection.py [hunk 2]
```diff
@@ -637,3 +637,45 @@ def scan_fn(carry, xy):
...
+    first_obs = obs_list[0]
+    coadd_im = jnp.zeros_like(first_obs.image)
+    coadd_noise = jnp.zeros_like(first_obs.image)
+    mask = jnp.zeros(coadd_im.shape, dtype=jnp.int32)
+    total_weight = jnp.zeros_like(first_obs.image)
+
+    for obs in obs_list:
+        coadd_im = coadd_im + obs.image * obs.weight
+        coadd_noise = coadd_noise + obs.noise * obs.weight
+        total_weight = total_weight + obs.weight
```
Comment (beckermr) [inline deep_field_metadetect/jaxify/jax_detection.py]: Weight maps are 1/variance so to add them we should be computing `1 / sum 1/w_i`.

Comment (beckermr) [reply deep_field_metadetect/jaxify/jax_detection.py]: Also this computation should include the scalar weight we use applied to the weight maps:

`1 / sum (w_i / (sum_k w_k))^2 / weightmap_i`

where `w_i / (sum_k w_k)` is the normalized weight per image.

File: deep_field_metadetect/jaxify/jax_metadetect.py [hunk 3]
```diff
@@ -338,3 +340,517 @@ def process_detection(
...
+        shears = DEFAULT_SHEARS
+
+    if detband_indices is None:
+        detband_indices = tuple(range(n_bands))
+
+    # shear images for each band
+    # Only process bands that will be used for measurements (detband_indices)
+    mcal_res_dict = {}
+    kinfo_dict = {}
+    # TODO: vmap this?
```
Comment (beckermr) [inline deep_field_metadetect/jaxify/jax_metadetect.py:500]: Yes we should vmap this.

File: deep_field_metadetect/jaxify/jax_metadetect.py [hunk 4]
```diff
@@ -338,3 +340,517 @@ def process_detection(
...
+
+        # Compute bmask_flags only for actual detections
+        valid_mask = det_flag == 1
+        ixc = jnp.where(valid_mask, (x_coords + 0.5).astype(int), 0)
+        iyc = jnp.where(valid_mask, (y_coords + 0.5).astype(int), 0)
+        bmask_flags = jnp.where(valid_mask, detobs.bmask[iyc, ixc], 0)
+        detections.append(detres)
+
+        # Process measurements for each band in detband_indices for every detection
+        for band_idx in detband_indices:
```
Comment (beckermr) [inline deep_field_metadetect/jaxify/jax_metadetect.py:559]: We should vmap over or at least bands within each shear?

**Review by beckermr — CHANGES_REQUESTED** [review]
The function looks good and I am glad to see so few changes to get the code to do this. 

Maybe we can break it up into smaller routines?

I also left some notes on the vmapping.

**Comment by beckermr** [comment]
We may want to delay the vmapping work to see how this works if we vmap over a set of cells.

**Review by beckermr — APPROVED** [review]
